yt_dlp/extractor/nytimes.py

```diff
@@ -124,139 +88,347 @@ class NYTimesIE(NYTimesBaseIE):
-            r'[Ee]pisode\s+(\d+)', episode, 'episode number', default=None))
+        # audio articles will have an url and no formats
+        url = traverse_obj(block, ('fileUrl', {url_or_none}))
+        if not formats and url:
+            formats.append({'url': url})
 
         return {
-            'id': remove_start(podcast_audio.get('target'), 'FT') or page_id,
-            'url': video_url,
-            'title': title,
-            'description': description,
-            'creator': track.get('credit'),
-            'series': podcast_title,
-            'episode': episode_title,
-            'episode_number': episode_number,
-            'duration': int_or_none(track.get('duration')),
+            **details,
+            'formats': formats,
+            'subtitles': subtitles
         }
 
     def _real_extract(self, url):
         page_id = self._match_id(url)
-
         webpage = self._download_webpage(url, page_id)
+        art_json = self._search_json(
+            r'window\.__preloadedData\s*=', webpage, 'media details', page_id,
+            transform_source=lambda x: x.replace('undefined', 'null'))['initialData']['data']['article']
+
+        common_info = {
+            'title': remove_end(self._html_extract_title(webpage), ' - The New York Times'),
+            'description': traverse_obj(art_json, (
+                'sprinkledBody', 'content', ..., 'summary', 'content', ..., 'text', {str}),
+                get_all=False) or self._html_search_meta(['og:description', 'twitter:description'], webpage),
+            'timestamp': traverse_obj(art_json, ('firstPublished', {parse_iso8601})),
+            'creator': ', '.join(
+                traverse_obj(art_json, ('bylines', ..., 'creators', ..., 'displayName')))  # TODO: change to 'creators' (list)
+        }
 
-        video_id = self._search_regex(
-            r'data-videoid=["\'](\d+)', webpage, 'video id',
-            default=None, fatal=False)
-        if video_id is not None:
-            return self._extract_video_from_id(video_id)
+        blocks = traverse_obj(art_json, (
+            'sprinkledBody', 'content', ..., ('ledeMedia', None),
+            lambda _, v: v['__typename'] in ('Video', 'Audio')))
 
-        podcast_data = self._search_regex(
-            (r'NYTD\.FlexTypes\.push\s*\(\s*({.+?})\s*\)\s*;\s*</script',
-             r'NYTD\.FlexTypes\.push\s*\(\s*({.+})\s*\)\s*;'),
-            webpage, 'podcast data')
-        return self._extract_podcast_from_json(podcast_data, page_id, webpage)
+        # more than 1 video in the article, treat it as a playlist
+        if len(blocks) > 1:
+            entries = []
+            for block in blocks:
+                entries.append(merge_dicts(
+                    self._extract_content_from_block(block), common_info, {
+                        'thumbnails': self._extract_thumbnails(traverse_obj(
+                            block, ('promotionalMedia', 'crops', ..., 'renditions', ...)))}))
```

Comment:
if applying the above 2 suggestions we can simplify like this (and remove below)
```suggestion
        entries = []
        for block in blocks:
            entries.append({
                **common_info,
                **self._extract_content_from_block(block),
            })

        if len(entries) > 1:
            return self.playlist_result(entries, page_id, **common_info)

        return {
            'id': page_id,
            **common_info,
            **self._extract_content_from_block(blocks[0]),
        }
```

Reply:
```python
            entries.append({
                **common_info,
                **self._extract_content_from_block(block),
            })
```

If the thumbnails are extracted in the ``common_info`` and in ``self._extract_content_from_block(block)`` the value  is ``None``, the `entry` value for thumbnails will be ``None``.

Reply:
Changed the suggested code to this:
```python
        for block in blocks:
            entries.append(merge_dicts(self._extract_content_from_block(block), common_info))
```

And adjusted the ``_extract_thumbnails`` to return `Ǹone` if empty.

```python
    def _extract_thumbnails(self, thumbs):
        return traverse_obj(thumbs, (lambda _, v: url_or_none(v['url']), {
            'url': 'url',
            'width': ('width', {int_or_none}),
            'height': ('height', {int_or_none}),
        })) or None
```

This returns the expected value for ``thumbnails`